File: in_Java/0566 Reshape the Matrix.java
class Solution {
    public int[][] matrixReshape(int[][] nums, int r, int c) {
        if(nums == null || nums.length == 0 || nums[0].length == 0) return nums;
        int row = nums.length;
        int col = nums[0].length; 
        if(row * col != r * c) return nums;
        int[] n = new int[r*c];
        int id = 0;
        for(int i = 0; i < row; i++){
            for(int j = 0; j < col; j++){
                n[id++] = nums[i][j];
            }
        }
        int[][] res = new int[r][c];
        for(int i = 0; i < r; i++){
            for(int j = 0; j < c; j++){
                res[i][j] = n[j+i*c];
            }
        }
        return res;
    }
}

File: in_Python/0566 Reshape the Matrix.py
class Solution(object):
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        row = len(nums)
        col = len(nums[0])
        if row * col != r * c:
            return nums
        alln = []
        for i in range(row):
            for j in range(col):
                alln.append(nums[i][j])
        res = []
        for i in range(r):
            cs = []
            for j in range(c):
                cs.append(0)
            res.append(cs)
        idx = 0
        for i in range(r):
            for j in range(c):
                res[i][j] = alln[idx]
                idx += 1
        return res